webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/checks/migration/CheckTableReportsExecutionTz.scala

```diff
@@ -0,0 +1,120 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+ *
+ *************************************************************************************
+ */
+
+import bootstrap.liftweb.*
+import com.normation.errors.IOResult
+import com.normation.rudder.db.Doobie
+import com.normation.zio.*
+import doobie.Update0
+import doobie.implicits.*
+import doobie.util.fragment.Fragment
+import zio.interop.catz.*
+
+/*
+ * When we created table ReportsExecution, we didn't use "timestamp with time zone" for column
+ * 'insertiondate' and 'compliancecomputationdate'
+ */
+class CheckTableReportsExecutionTz(
+    doobie: Doobie
+) extends BootstrapChecks {
+
+  import CheckTableReportsExecutionTz.*
+  import doobie.*
+
+  override def description: String =
+    "Check if database table  ReportsExecution has timestamps with time zone and correct it if not"
+
+  def addTz(table: String, column: String, overrideChange: Option[Fragment] = None): IOResult[Unit] = {
+
+    transactIOResult(s"Error when adding time zone to column ${table}.${column}")(xa =>
+      query(table, column, overrideChange).run.transact(xa)
+    ).unit
+  }
+
+  def migrationProg(table: String, overrideChange: Option[Fragment] = None): IOResult[Unit] = {
+    for {
+      _ <- addTz(table, insertionDateColumn, overrideChange)
+      _ <- addTz(table, complianceComputationDateColumn, overrideChange)
+    } yield ()
+  }
+
+  override def checks(): Unit = {
+    // Actually run the migration async to avoid blocking for that.
+    // There is no need to have it sync.
+    migrationProg(reportsExecution)
+      .catchAll(err => BootstrapLogger.error(s"Error when trying to add time zone to ReportsExecution table: ${err.fullMsg}"))
+      .forkDaemon
+      .runNow
+  }
+
+}
+
+object CheckTableReportsExecutionTz {
+  val reportsExecution:                String = "reportsexecution"
+  val insertionDateColumn:             String = "insertiondate"
+  val complianceComputationDateColumn: String = "compliancecomputationdate"
+
+  // overrideChange is only used for tests
+  def query(table: String, column: String, overrideChange: Option[Fragment] = None): Update0 = {
+    def toQuotedFr(s: String) = Fragment.const("'" + s + "'")
+    def toFr(s:       String) = Fragment.const(s)
+
+    val alter: Fragment = fr"ALTER TABLE " ++ toFr(table) ++ fr" ALTER COLUMN " ++ toFr(
+      column
+    ) ++ fr" TYPE TIMESTAMP WITH TIME ZONE USING " ++ toFr(column) ++ fr" AT TIME ZONE 'UTC';"
+
+    val sql = {
+      fr"""
+        DO $$$$ BEGIN
+          IF NOT EXISTS (
+            select column_name, data_type
+            from information_schema.columns
+            where table_name = """ ++ toQuotedFr(table) ++ fr" and column_name = " ++ toQuotedFr(
+        column
+      ) ++ fr""" and data_type = 'timestamp with time zone'
```

Comment:
The table and column does not need to be constant fragment but can simply be interpolated, since they are standard parameterized arguments for the query, and not part of the SQL DML syntax,
for instance : https://github.com/Normation/rudder/blob/a900c33686260e9b935f49e3983e9caa44dad193/webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/checks/migration/DbCommonMigration.scala#L13 (the License header is missing from that file :see_no_evil:)

Reply:
Added and fixed the quoting convention - we always use `${}`